pimd/pim6_mld.c

```diff
@@ -444,6 +444,13 @@ static void gm_sg_update(struct gm_sg *sg, bool has_expired)
 	}
 
 	if (desired == GM_SG_NOINFO) {
+		/*
+		 * If oil is still present then get ride of it or we will leak
+		 * this data structure.
+		 */
+		if (sg->oil)
+			pim_channel_oil_del(sg->oil, __func__);
+
```

Comment:
Is there a need to backport this?

Reply:
I don't know, I only saw this issues after trying the setup with embedded RP. Normally the channels are only created when there are RPs configured, but in embedded-RP case it is created on stimulus reception (e.g. MLD or PIM packets) - increases refcount - and gone when embedded-RP or state is removed.